### ISSUE TITLE
Report outcome/duration of splits when they occur in /verificationLogger:trx content

### ISSUE THREAD

**Issue by robin-aws** [dafny-lang/dafny#1624]
Depends on https://github.com/boogie-org/boogie/pull/461

The data recorded by the `/verificationLogger:trx` option can help to identify Boogie procedures that take much longer to verify, which we hypothesize in #1582 are therefore more likely to fail to verify in the future as code is changed. However, if a procedure is verified in multiple Boogie verification conditions because of directives like `assert {:split_here} true` or `assert {:focus} true`, those "splits" become the actual unit of verification. This means that Dafny codebases should pay attention to the verification time of individual splits whenever splitting occurs, instead of the total time to verify all splits for a particular procedure (especially since they may even be verified in parallel).

Unfortunately there isn't a good way to identify an individual Boogie split yet, even at the Boogie level, which makes it challenging to provide a good user experience (and I'm going to cut a separate issue for this as it is a deeper gap to address). As incremental progress, I propose just using the "split number" to disambiguate, using identifiers such as `CheckWellformed$$MyModule.__default.MyMethod$$7`. The exact format of the identifiers in the TRX output is already flagged as experimental in the help text for the option, and this feature wouldn't affect that status.
